src/data/set/finite.lean

```diff
@@ -513,8 +513,13 @@ lemma univ_finite_iff_nonempty_fintype :
   (univ : set α).finite ↔ nonempty (fintype α) :=
 ⟨λ h, ⟨fintype_of_finite_univ h⟩, λ ⟨_i⟩, by exactI finite_univ⟩
 
-lemma finite.to_finset_insert [decidable_eq α] {a : α} {s : set α} (hs : s.finite) :
-  (hs.insert a).to_finset = insert a hs.to_finset :=
+@[simp] lemma finite.to_finset_singleton {a : α} (ha : ({a} : set α).finite := finite_singleton _) :
+  ha.to_finset = {a} :=
+finset.ext $ by simp
+
+@[simp] lemma finite.to_finset_insert [decidable_eq α] {s : set α} {a : α}
```

Comment:
I don't think this lemma is an improvement over the removed one; I suspect this is a case where we want both versions, one for each direction of rewrite

Reply:
Same thought here, so I restored it but unsimped and primed.